templates/etcd_network_definition.erb

```diff
@@ -1,25 +1,24 @@
 {
     "Network": "<%= scope['flannel::network'] %>"<% -%>
-<% if @subnetlen then -%>,
+<% if scope['flannel::subnetlen'] then -%>,
```

Comment:
This will not work in all cases. For an explanation see: http://stackoverflow.com/questions/17096934/check-if-a-variable-is-undef-in-puppet-template.

You can check how the guys from redhat check for undefine: https://github.com/Katello/puppet-pulp/blob/master/templates/docker_importer.json.erb